oneflow/core/graph/stream_index_getter_registry_manager.h

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2020 The OneFlow Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#ifndef ONEFLOW_CORE_GRAPH_STREAM_INDEX_GETTER_REGISTRY_MANAGER_H_
+#define ONEFLOW_CORE_GRAPH_STREAM_INDEX_GETTER_REGISTRY_MANAGER_H_
+
+#include "oneflow/core/common/util.h"
+#include "oneflow/core/common/device_type.pb.h"
+#include "oneflow/core/graph/stream_index_getter_registry.h"
+
+namespace oneflow {
+
+struct CustomHash {
+  std::size_t operator()(const std::pair<DeviceType, TaskType>& pair) const {
+    return (static_cast<std::size_t>(pair.first) << 10) + static_cast<std::size_t>(pair.second);
+  }
+};
+
+template<typename Value>
+using StreamIndexKeyMap = std::unordered_map<std::pair<DeviceType, TaskType>, Value, CustomHash>;
+
+class StreamIndexGetterRegistryManager final {
+ private:
+  StreamIndexGetterRegistryManager() {}
+
+ public:
+  OF_DISALLOW_COPY_AND_MOVE(StreamIndexGetterRegistryManager);
+
+ public:
+  static StreamIndexGetterRegistryManager& Get();
+
+  StreamIndexKeyMap<StreamIndexGetterFn>& StreamIndexGetterFuncs();
+
+ private:
+  StreamIndexKeyMap<StreamIndexGetterFn> stream_index_getter_funcs_;
+};
+
+#define REGISTER_COMP_TASK_NODE_STREAM_INDEX_GETTER(device_type, task_type)                      \
```

Comment:
感觉 OneFlow 里公共宏的命名一般不用缩写，特别是大部分字母没有缩写，缩写其中一个，显得有点奇怪。
```suggestion
#define REGISTER_COMPUTE_TASK_NODE_STREAM_INDEX_GETTER(device_type, task_type)                      \
```